### ISSUE TITLE
Add new entry next in the queue with /qan

### ISSUE THREAD

**Issue by girlmaya** [Syncplay/syncplay#702]
I often find myself adding a video to the queue with /qa, and then dragging the video up to the point in the queue below the current video. A /qan command would simplify the process. 
A /qans command (queue add next and select) would also be useful, though the command would probably need a better name.

**Comment by Et0h** [comment]
These additional commands seem reasonable to me and shouldn't be too hard to code. I actually half thought we might already had a queue next feature so had to double check (you can see I mostly use the GUI) because it seems like a common enough use case for CLI users. The downside is of course that it adds yet more commands which could be a bit daunting to non-power users, but they will mostly use the GUI rather than CLI in any case.

Based on the other commands I think /qan and /quans seem like the logical names even if they are a bit cumbersome as we use '**qa** for **queue add**, 'qa**s**' for queue add and **select**' and 'q**n**' for select **next** in queue and so this would just be re-using the existing meanings.

**Comment by girlmaya** [comment]
I agree it's a bit tricky and don't really have a preference on the options you gave. But here's another option:

If you were to rework the queue commands entirely, we could have `/queue add <file>` (`/q a <file>`), with additional flags (for only the `add` subcommand) like `/queue add --next <file>` (`/q an <file>`) and `/queue add --select <file>` (`/q as <file>`). These could be able to be combined so that the user would be able to use `/q ans <file>` or `/q asn <file>` to add a video next in the queue, and select it for playback. Other existing queue management commands like `/ql`, `/qs`, `/qn`, `/qd` could get their own `/queue` subcommands (list, select/sel, next, delete/del), possibly even keeping the old commands as aliases. That way the only old command that would break is `/queue`, though I'd guess most people (including me) typically use `/qa`, which currently does the same thing.

I assume (without having looked at the code) that this would involve a rework of the command system. I don't think this is the absolute necessary solution, nor am I sure it is even better than simply adding `/qan` and `/qans` (might be a bit too convoluted). Just a suggestion.

FYI I am not a CLI user - I use the GUI but add to the queue with `/qa` (usually from youtube).

**Comment by Et0h** [comment]
If there were any changes to rationalise things then I'd still probably want the old commands to still work for backwards compatibility reasons.

I will tag some people who have made commits or issue comments relating to CLI to give them an opportunity to comment on these proposed changes: @crabdancing @gospodin55 @rtix @csandras05 @odrling @xNinjaKittyx @luke-jr

**Comment by odrling** [comment]
> If you were to rework the queue commands entirely, we could have `/queue add <file>` (`/q a <file>`), with additional flags (for the `add` subcommand) like `/queue add --next <file>` (`/q an <file>`) and `/queue add --select <file>` (`/q as <file>`). These should be able to be combined so that the user would be able to use `/q ans <file>` or `/q asn <file>` to add a video next in the queue, and select it for playback. Other existing queue management commands like `/ql`, `/qs`, `/qn`, `/qd` could get their own `/queue` subcommands (list, select/sel, next, delete/del), possibly even keeping the old commands as aliases. That way the only old command that would break is `/queue`, though I'd guess most people (including me) typically use `/qa`, which currently does the same thing.
> 
> I assume (without having looked at the code) that this would involve a rework of the command system. I don't think this is the absolute necessary solution, nor am I sure it is even better than simply adding `/qan` and `/qans` (might be a bit too convoluted). Just a suggestion.

While it feels instinctively right to implement commands this way, I feel like in practice most users would still prefer to use short aliases in Syncplay (if they even use them) because having more characters to type means you have more chances to make a typo (and it also just takes more time).

That said I think a good thing that could come from having longer commands (without necessarily implementing subcommands and flags) is user defined aliases. These aliases could still default to mimicking the commands we have currently to keep backwards compatibility, and then it would be possible e.g. to alias the `/t` command to `/r` (which then would replace the `/r` alias) and `/qa` to `/a`.

**Comment by girlmaya** [comment]
> While it feels instinctively right to implement commands this way, I feel like in practice most users would still prefer to use short aliases in Syncplay (if they even use them) because having more characters to type means you have more chances to make a typo (and it also just takes more time).

Does `/q ans` seem too cumbersome? Is it because of the additional space, or is there a better sequence of characters that remains somewhat intuitive?

**Comment by odrling** [comment]
Well, that was more about `/queue add --next --select` and alikes.
But IOW, why bother with implementing this subcommand and flags system if we don’t think anyone will use them that way?

If it’s just so `/q ans` and `/q asn` (with or without spaces) have the same semantics that sounds like more work for something that, in practice, seems very similar to what we already have (except that we don’t have two commands doing the same thing).

**Comment by girlmaya** [comment]
I agree the longer form flags would rarely be used, but it makes the most sense that way in my head

**Comment by girlmaya** [comment]
I agree the longer form flags would rarely be used. I do think `/queue` subcommands should be implemented though instead of having a new command with every variation of adding/removing
